javalib/src/main/scala/java/util/Base64.scala

```diff
@@ -0,0 +1,390 @@
+      val dstLen = dstLength(src.length)
+      if (dst.length < dstLen)
+        throw new IllegalArgumentException("Output byte array is too small for encoding all input bytes")
+      doEncode(src, dst, dstLen)
+    }
+
+    def encode(buffer: ByteBuffer): ByteBuffer = {
+      val result = ByteBuffer.allocate(dstLength(buffer.remaining()))
+      doEncode(buffer, result)
+      result.flip()
+      result
+    }
+
+    def encodeToString(src: Array[Byte]): String = new String(encode(src), StandardCharsets.ISO_8859_1)
+
+    def withoutPadding(): Encoder = if (withPadding) new Encoder(table, lineLength, lineSeparator, false) else this
+
+    def wrap(os: OutputStream): OutputStream =
+      new EncodingOutputStream(os, table, lineLength, lineSeparator, withPadding)
+
+    private def doEncode(src: Array[Byte], dst: Array[Byte], dstLength: Int): Int =
+      doEncode(ByteBuffer.wrap(src), ByteBuffer.wrap(dst, 0, dstLength))
+
+    // dst position must always be 0 here
+    private def doEncode(src: ByteBuffer, dst: ByteBuffer): Int = {
+      val length = src.remaining()
+      var currentLine = 0
+
+      @inline
+      def encode(a: Byte, b: Byte, c: Byte): Unit = {
+        val bits = (a & 0xff) << 16 | (b & 0xff) << 8 | (c & 0xff)
+        dst.put(table((bits >>> 18) & 0x3f))
+        dst.put(table((bits >>> 12) & 0x3f))
+        if(dst.hasRemaining) dst.put(table((bits >>> 6) & 0x3f))
+        if(dst.hasRemaining) dst.put(table(bits & 0x3f))
+        currentLine += 4
+        if(lineSeparator.length > 0 && lineLength > 0 && currentLine == lineLength && dst.hasRemaining) {
+          lineSeparator.foreach(dst.put)
+          currentLine = 0
+        }
+      }
+
+      while(src.remaining() >= 3) encode(src.get(), src.get(), src.get())
+
+      (3 - length % 3) % 3 match {
+        case 0 =>
+        case 1 =>
+          encode(src.get(), src.get(), 0)
+          if(withPadding) {
+            dst.position(dst.position()-1)
+            dst.put('='.toByte)
+          }
+        case 2 =>
+          encode(src.get(), 0, 0)
+          if(withPadding) {
+            dst.position(dst.position()-2)
+            dst.put('='.toByte)
+            dst.put('='.toByte)
+          }
+      }
+      dst.position()
+    }
+
+    private def dstLength(srcLength: Int): Int = {
+      val withPad = ((srcLength + 2) / 3) * 4
+      val toRemove = if (withPad == 0 || withPadding) 0 else (3 - (srcLength % 3)) % 3
+      val withoutEndLines = withPad - toRemove
+      withoutEndLines + (if (lineLength > 0) ((withoutEndLines-1) / lineLength) * lineSeparator.length else 0)
+    }
+  }
+
+  // --------------------------------------------------------------------------
+
+  private class EncodingOutputStream(out: OutputStream,
+                                     table: ByteTable,
+                                     ll: Int,
+                                     ls: Array[Byte],
+                                     withPadding: Boolean) extends OutputStream {
```

Comment:
It's probably easier if you extend `FilterOutputStream`. It gives you some correct default implementations, such as `flush()`.